client/src/Molecules.elm

```diff
@@ -226,46 +389,180 @@ level1Decoder =
+          )
+        , ( "bsmb1_overhang"
+          , Encode.string <|
+                showBsmb1Overhang <|
+                    Maybe.withDefault InvalidBsmb1 backbone.bsmb1Overhang
+          )
+        , ( "sequence", Encode.string "NNNNNNNNNNNNNNNNN---NNNNNNN" ) -- TODO: We do need the sequence!!!
+        , ( "bacterial_strain"
+          , Encode.string <|
+                Maybe.withDefault "" backbone.bacterialStrain
+          )
+        , ( "responsible", Encode.string <| backbone.responsible )
+        , ( "group", Encode.string <| backbone.group )
+        , ( "selection", Encode.string <| Maybe.withDefault "" backbone.selection )
+        , ( "notes", Encode.string <| Maybe.withDefault "" backbone.notes )
+        , ( "REase_digest", Encode.string <| Maybe.withDefault "" backbone.reaseDigest )
+        , ( "date", Encode.string <| Maybe.withDefault "" backbone.date )
+        , ( "vector_type", Encode.string <| Maybe.withDefault "" backbone.vectorType )
+        ]
+
+
+level0Encoder : Level0 -> Encode.Value
+level0Encoder level0 =
+    Encode.object
+        [ ( "name", Encode.string level0.name )
+        , ( "location", Encode.int level0.location )
+        , ( "bsa1_overhang", Encode.string <| showBsa1Overhang <| level0.bsa1Overhang )
+        , ( "bacterial_strain", Encode.string <| Maybe.withDefault "" level0.bacterialStrain )
+        , ( "responsible", Encode.string <| level0.responsible )
+        , ( "group", Encode.string <| level0.group )
+        , ( "selection", Encode.string <| Maybe.withDefault "" level0.selection )
+        , ( "cloning_technique", Encode.string <| Maybe.withDefault "" level0.cloningTechnique )
+        , ( "is_BsmB1_free", Encode.string <| isBsmb1FreeToString level0.isBsmb1Free )
+        , ( "notes", Encode.string <| Maybe.withDefault "" level0.notes )
+        , ( "REase_digest", Encode.string <| Maybe.withDefault "" level0.reaseDigest )
+        , ( "date", Encode.string <| Maybe.withDefault "" level0.date )
+        , ( "genbank_content", Encode.string <| Maybe.withDefault "" level0.genbankContent )
+        ]
+
+
+levelNEncoder : Level1 -> Encode.Value
+levelNEncoder level1 =
+    Encode.object
+        [ ( "name", Encode.string level1.name )
+        , ( "location", Encode.int level1.location )
+        , ( "bsmb1_overhang"
+          , Encode.string <|
+                showBsmb1Overhang <|
+                    Maybe.withDefault InvalidBsmb1 level1.bsmb1Overhang
+          )
+        , ( "responsible", Encode.string <| level1.responsible )
+        , ( "group", Encode.string <| level1.group )
+        , ( "notes", Encode.string <| Maybe.withDefault "" level1.notes )
+        , ( "sequenceLength", Encode.int level1.sequenceLength )
+        , ( "genbank_content", Encode.string <| Maybe.withDefault "" level1.genbankContent )
+        ]
+
+
 parseBsa1Overhang : String -> Maybe Bsa1Overhang
 parseBsa1Overhang str =
     case str of
```

Comment:
```suggestion
    case String.replace "_" "" str of
```

This allows you to keep only the "XY" cases and remove all the duplications from "X__Y"